### DISCUSSION TITLE
Template for generated function tests?

### DISCUSSION THREAD

**Discussion by majohn-r** [golang/vscode-go#3333] [Help]
Is there a template for the generated function tests? I'd like to make a simple tweak to the tests, so that instead of tests being a slice of the struct holding the test parameters (one of which is called name), it's a map of the structs, keyed by strings, and then the test iteration is

for name, tt := range tests {
		t.Run(name, func(t *testing.T) { ... etcetera

this prevents the test cases from being run in the same order, every time, which hides issues where one test case modifies the environment so that a subsequent test case succeeds where, maybe, it shouldn't.

**Comment by butuzov** [comment]
Hi. You can try to use same old  ‘gotests’, however changes added to codebase wasn’t ever released (and I have already lost hope they will be released ever again). so, what you can do is:
- build gotests from develop branch.
- used named as template for code generation.

cheers from Ukraine
